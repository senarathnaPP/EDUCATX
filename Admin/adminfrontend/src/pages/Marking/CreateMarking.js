import React, { useEffect, useState } from 'react';
import './../Markings.css';
import axios from "axios"
import 'react-toastify/dist/ReactToastify.css';
import {toast} from 'react-toastify';
import AdminNavbar from '../../components/AdminNavbar/adminNavbar';
import Footer from '../../components/Footer/Footer';
import {Link } from "react-router-dom";

toast.configure()
const CreateMarking =() =>{
    const [markingId, setMarkingId] = useState('');
    const [title, setTitle] = useState('');
    const [category, setCategory] = useState('');
    const [description, setDescription] = useState('');
    const [updatedDate, setUpdatedDate] = useState('');
    
         
  //  function setUpdatedDate(object){               
           
  //  }


  const notify1 = () =>{
        toast.success('Marking Scheme is created Successfully !', {position: toast.POSITION.TOP_CENTER})
  };

  const notify2 = () =>{
    toast.error('Marking Scheme is not added !', {position: toast.POSITION.TOP_CENTER})
   };

  const changeOnClick = e => {
    e.preventDefault();

    /** */
    // const valid = this.validate();

    const markings = {
        markingId,
        title,
        category,
        description,
        updatedDate

    };

    setMarkingId("");
    setTitle("");
    
    if((category >= 'a' && category <= 'z') || (category >= 'A' && category <= 'Z'))
    {
    setCategory("");
    }
    else{
      console.log("Bad Category");
      alert("Category should be a character")
      return false;
    }


    setDescription("");

    // Date shuold be greater than or equal today's date
    var date = new Date();
        console.log("Date: "+date.getDate()+
          "/"+(date.getMonth()+1)+
          "/"+date.getFullYear() );

          var mDate = new Date(updatedDate);
          console.log("MyDate: "+mDate.getDate()+
            "/"+(mDate.getMonth()+1)+
            "/"+mDate.getFullYear() );

          if(date.getFullYear() < mDate.getFullYear()){
            console.log("Good 1");
            setUpdatedDate();
          }
  
          else if(date.getFullYear() == mDate.getFullYear() && (date.getMonth()+1) < (mDate.getMonth()+1)){
            console.log("Good 2");
            setUpdatedDate();
          }

          else if((date.getMonth()+1) == (mDate.getMonth()+1) && date.getDate() <= mDate.getDate()){
            console.log("Good 3");    
            setUpdatedDate();
          }

          else {
            console.log("Bad");
            alert("Date is invalid! It should be greater than or equal to current date")
            return false;
          }
     

     

    axios.post("http://localhost:8000/api/admin/marking/create", markings)
        .then(res => {
          notify1("");
          console.log(res.data)
          
        }
          )
        // .catch(err =>{
        //     console.log(err);
        // });
        .catch(error=> {
          alert("Some Fields are Empty")
          notify2("");
          });  
     

  }
    

    return (
      <div data-testid="marking">
        <div>
             <AdminNavbar/> 
             </div>
             <br></br>
       
        <div className='containerA'> 
        <h2>Create New Marking Scheme</h2>
        <form onSubmit={changeOnClick} encType="multipart/form-data">
    <div className="form-group">
    <label htmlFor="markingId">ID</label>
    <input 
        type="text" 
        value={markingId}
        onChange={e => setMarkingId(e.target.value)}
        className="form-control" 
        placeholder="Enter Marking Scheme ID"/>
  </div>


  <div className="form-group">
    <label htmlFor="title">Title</label>
    <input 
        type="text" 
        value={title}
        onChange={e => setTitle(e.target.value)}
        className="form-control" 
        placeholder="Enter the title"/>
  </div>

  {/* <div className="form-group">
    <label htmlFor="category">Category</label>
    <input 
    type="text" 
    value={category}
    onChange={e => setCategory(e.target.value)}
    className="form-control" 
    placeholder="Enter the category"/>
  </div> */}
<div className="form-group">
<label htmlFor="category">Category</label>
<select class="form-select" aria-label="Default select example" value={category}
    onChange={e => setCategory(e.target.value)}>
  <option selected>Select Category</option>
  <option >A</option>
  <option >B</option>
  <option >C</option>
  <option >D</option>
</select>
</div>

  <div className="form-group">
    <label htmlFor="description">Marking Scheme</label>
    <textarea 
      value={description}
    className="form-control" 
    onChange={e => setDescription(e.target.value)}
    rows="7"></textarea>
  </div>

  <div className="form-group">
    <label htmlFor="updatedDate">Updated Date</label>
    <input 
    type="date" 
    value={updatedDate}
    className="form-control"
    onChange={e => setUpdatedDate(e.target.value)} 
    placeholder="Enter the relevant updated date"/>
  </div>
  <ul>
  <button type="submit" className="buttonAdd" >CREATE</button>

  <Link to="/viewMarkings" 
  >
  <button className="buttonAdd" >
    BACK TO LIST
  </button>
  </Link>

</ul>
</form>
</div>
            <div>
              <br></br>
             <Footer/> 
             </div>
             
</div>
    )
};

export default CreateMarking;