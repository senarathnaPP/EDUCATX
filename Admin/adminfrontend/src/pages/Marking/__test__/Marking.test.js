import React from 'react'; 
import ReactDOM from 'react-dom';
import CreateMarking from './../CreateMarking';
import { BrowserRouter } from 'react-router-dom';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";


import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
    <BrowserRouter>
     <CreateMarking></CreateMarking>
     </BrowserRouter>, div)
})

it("renders Create Marking correctly", () => {
    const {getByTestId} = render(<BrowserRouter><CreateMarking><div><h2>"Create New Marking Scheme"</h2></div></CreateMarking></BrowserRouter>)
    expect(getByTestId('marking')).toHaveTextContent("Create New Marking Scheme")
})

// it("renders Create Marking correctly", () => {
//     const {getByTestId} = render(<BrowserRouter><CreateMarking><div><h2>"DISPLAY Create New Marking Scheme"</h2></div></CreateMarking></BrowserRouter>)
//     expect(getByTestId('marking')).toHaveTextContent("Create New Marking Scheme")
// })

// //This will convert this to like a virtual DOM object
it("matches snapshot", () =>{
    const tree= renderer.create(<BrowserRouter><CreateMarking><div><form><label>"ID"</label></form></div></CreateMarking></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
    //It looks for folder called snapshot
} )