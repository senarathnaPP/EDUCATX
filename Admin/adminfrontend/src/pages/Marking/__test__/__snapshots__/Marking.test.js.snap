// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="marking"
>
  <div>
    <nav
      className="navbar"
    >
      <div
        className="navbar-container"
      >
        <a
          className="navbar-logo"
          href="/home"
          onClick={[Function]}
        >
          <img
            alt=""
            class=""
            src="logo1.png"
            style={
              Object {
                "width": 60,
              }
            }
          />
            ADMIN PANEL
        </a>
        <div
          className="menu-icon"
          onClick={[Function]}
        />
        <ul
          className="nav-menu"
        />
        <a
          href="/login"
        >
          <a
            className="btn--outline"
            href="/sign up"
            onClick={[Function]}
          >
            <button
              className="btn \${checkButtonStyle} \${checkButtonSize}"
            >
               LOG OUT
            </button>
          </a>
           
        </a>
      </div>
    </nav>
  </div>
  <br />
  <div
    className="containerA"
  >
    <h2>
      Create New Marking Scheme
    </h2>
    <form
      encType="multipart/form-data"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="markingId"
        >
          ID
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter Marking Scheme ID"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter the title"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter the category"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="description"
        >
          Marking Scheme
        </label>
        <textarea
          className="form-control"
          onChange={[Function]}
          rows="7"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="updatedDate"
        >
          Updated Date
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter the relevant updated date"
          type="date"
          value=""
        />
      </div>
      <ul>
        <button
          className="buttonAdd"
          type="submit"
        >
          CREATE
        </button>
        <a
          href="/viewMarkings"
          onClick={[Function]}
        >
          <button
            className="buttonAdd"
          >
            BACK TO LIST
          </button>
        </a>
      </ul>
    </form>
  </div>
  <div>
    <br />
    <div
      className="footer-container"
    >
      <section
        className="footer-subscription"
      >
        <p
          className="footer-subscription-heading"
        >
          Research Management Admin Panel
        </p>
        <p
          className="footer-subscription-text"
        >
          Manage all of your work in one place: Tasks, Docs, Chat, Goals, & more.
        </p>
        <div
          className="input-areas"
        >
          <form>
            <input
              className="footer-input"
              name="email"
              placeholder="Your Email"
              type="email"
            />
            <div
              className="btn--outline"
            >
              <a
                className="btn-mobile"
                href="/sign up"
                onClick={[Function]}
              >
                <button
                  className="btn \${checkButtonStyle} \${checkButtonSize}"
                >
                  Subscribe
                </button>
              </a>
            </div>
          </form>
        </div>
      </section>
      <div
        class="footer-links"
      >
        <div
          className="footer-link-wrapper"
        >
          <div
            class="footer-link-items"
          >
            <h3>
              About Research
            </h3>
            <a
              href="/sign-up"
              onClick={[Function]}
            >
              How it works
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Tools
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Careers
            </a>
          </div>
          <div
            class="footer-link-items"
          >
            <h3>
              Tasks
            </h3>
            <a
              href="/"
              onClick={[Function]}
            >
              User Management
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Panel Members 
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Roles 
            </a>
          </div>
        </div>
        <div
          className="footer-link-wrapper"
        >
          <div
            class="footer-link-items"
          >
            <h3>
              Creations
            </h3>
            <a
              href="/"
              onClick={[Function]}
            >
              Submission Types
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Marking Schemes
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Templates
            </a>
          </div>
          <div
            class="footer-link-items"
          >
            <h3>
              Quick Links
            </h3>
            <a
              href="/"
              onClick={[Function]}
            >
              Blogs
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              QA
            </a>
            <a
              href="/"
              onClick={[Function]}
            >
              Contact us
            </a>
          </div>
        </div>
      </div>
      <section
        class="social-media"
      >
        <div
          class="social-media-wrap"
        >
          <div
            class="footer-logo"
          >
            <a
              className="social-logo"
              href="/"
              onClick={[Function]}
            >
              <img
                alt=""
                class="logo"
                src="logo1.png"
                style={
                  Object {
                    "width": 50,
                  }
                }
              />
                EDUCATX Admin Panel
            </a>
          </div>
          <small
            class="website-rights"
          >
            EDUCATX Admin Panel © 2022
          </small>
        </div>
      </section>
    </div>
  </div>
</div>
`;
